lib/composite_cache_store.rb

```diff
@@ -4,120 +4,132 @@
 require_relative "composite_cache_store/version"
 
 class CompositeCacheStore
-  DEFAULT_OUTER_OPTIONS = {
+  DEFAULT_LAYER_1_OPTIONS = {
     expires_in: 5.minutes,
     size: 16.megabytes
   }
 
-  DEFAULT_INNER_OPTIONS = {
+  DEFAULT_LAYER_2_OPTIONS = {
     expires_in: 1.day,
     size: 32.megabytes
   }
 
-  attr_reader :outer_cache_store, :inner_cache_store
-
-  alias_method :outer, :outer_cache_store
-  alias_method :inner, :inner_cache_store
+  attr_reader :layers
+  delegate_missing_to :"layers.last"
 
   # Returns a new CompositeCacheStore instance
-  # - inner_cache_store: An ActiveSupport::Cache::Store instance to use for the inner cache store (typically remote)
-  # - outer_cache_store: An ActiveSupport::Cache::Store instance to use for the outer cache store (typically local)
-  def initialize(options = {})
-    options ||= {}
-
-    @inner_cache_store = options[:inner_cache_store]
-    @inner_cache_store = ActiveSupport::Cache::MemoryStore.new(DEFAULT_INNER_OPTIONS) unless inner.is_a?(ActiveSupport::Cache::Store)
+  def initialize(*layers)
+    if layers.blank?
+      layers << ActiveSupport::Cache::MemoryStore.new(DEFAULT_LAYER_1_OPTIONS)
+      layers << ActiveSupport::Cache::MemoryStore.new(DEFAULT_LAYER_2_OPTIONS)
+    end
 
-    @outer_cache_store = options[:outer_cache_store]
-    @outer_cache_store = ActiveSupport::Cache::MemoryStore.new(DEFAULT_OUTER_OPTIONS) unless outer.is_a?(ActiveSupport::Cache::Store)
+    layers.freeze
+    @layers = layers
   end
 
   def cleanup(...)
-    outer.cleanup(...)
-    inner.cleanup(...)
+    layers.each { |store| store.cleanup(...) }
   end
 
   def clear(...)
-    outer.clear(...)
-    inner.clear(...)
+    layers.each { |store| store.clear(...) }
   end
 
   def decrement(...)
-    outer.decrement(...)
-    inner.decrement(...)
+    layers.each { |store| store.decrement(...) }
   end
 
   def delete(...)
-    outer.delete(...)
-    inner.delete(...)
+    layers.each { |store| store.delete(...) }
   end
 
   def delete_matched(...)
-    outer.delete_matched(...)
-    inner.delete_matched(...)
+    layers.each { |store| store.delete_matched(...) }
   end
 
   def delete_multi(...)
-    outer.delete_multi(...)
-    inner.delete_multi(...)
+    layers.each { |store| store.delete_multi(...) }
   end
 
   def exist?(...)
-    outer.exist?(...) || inner.exist?(...)
+    layers.each do |store|
```

Comment:
could this be simplified with `layers.any?` ?

Reply:
That's what I thought initially, but I'm pretty sure that any? will hit all cache layers before returning. I want to return as early as possible which is why I changed to this impl.

Reply:
In that case, maybe you could use `find`: https://devdocs.io/ruby~3.2/enumerable#method-i-find

but that's nitpicking for sure.